caffe2/python/tutorials/py_gen/MNIST.py

```diff
@@ -152,48 +174,61 @@ def AddInput(model, batch_size, db, db_type):
+# Now we are going to construct our own model. The input will be our data blob ($X$), and the output will be a vector of length 10 containing the network's prediction on each of the 10 possible digits in MNIST.
 # 
-# Relu(x) = x if x > 0 else 0
+# In this Multi-layer Perceptron (MLP) architecture, the ReLU activation function is going to be used and is defined as:
 # 
-# Each layer of an MLP is just matrix multiplication with a bias plus an activation function. In our case, with ReLU activation, that is:
+# $$Relu(x) =\begin{cases}x & if x > 0\\ 0 & otherwise\end{cases}$$
 # 
-# layer1 = Relu(X * W1^T + b1)
+# Recall, each layer of an MLP is just matrix multiplication between some input data and a weight matrix ($W$), then an addition with a bias ($b$).  In machine learning this is accomplished with a fully-connected layer, which is aptly named [FC](https://caffe2.ai/docs/operators-catalogue.html#fc) in Caffe2. The result of this multiplication is then passed to a non-linear activation function (ReLU). The output of each activation function is passed to the next layer as the input data and this process is repeated through the layers as follows:
 # 
-# layer2 = Relu(layer1 * W2^T + b2)
+# $ {layer_1 = Relu(XW_1^T + b_1) } $
 # 
-# ...
+# $ {layer_2 = Relu(layer_1W_2^T + b_2) } $
 # 
-# Ultimately we will use the Softmax operator to convert scores for each of the digits to probabilities. So (p_0 + ... + p_9) = 1.0 and 0 <= p_i <= 1.0. 
+# $ {...} $
 # 
-# There are more detailed MLP explanations online. A good example is [here](http://deeplearning.net/tutorial/mlp.html).
+# Once the data has propagated through the network in the forward pass, we will use the [Softmax](https://caffe2.ai/docs/operators-catalogue.html#softmax) operator to convert scores for each of the digits to probabilities. As a rule, the scores sum to one (i.e. $p_0 + ... + p_9 = 1.0$) and no single score is greater than or equal to one (i.e. $0 <= p_i <= 1.0$). A more detailed description of MLP can be found [here](http://deeplearning.net/tutorial/mlp.html).
 # 
-# In this function we are going to use Brew for the second time. Please refer to the explanation given above. When below we call brew.fc(model, layer, ...) under the hood the following happens. FC operator is going to be added to model.net by calling model.net.FC([layer, W, b], ...). Where W and b are the weight and the bias of this fully connected layer (output = layer * W^T + b). Initially, we get W and b by adding their initialization into model.param_init_net. All of these is happening under the hood. You could just use Brew! :) 
+# In this function we are going to use Brew for the first time. Under the hood, when we call `brew.fc(model, layer, ...)` the following happens:
+# 
+# - The FC operator is added to `model.net` by calling `model.net.FC([layer, W, b], ...)`
+# - The initializations for $W$ and $b$ are added to the `model.param_init_net`
+# 
+# Hopefully, this makes the convenience of brew is clear! 
```

Comment:
Nit: Hopefully, this makes the convenience of brew is clear!  -> Hopefully, this makes the convenience of brew clearer!